app/server/lib/DocApi.ts

```diff
@@ -2320,6 +2325,41 @@ export class DocWorkerApi {
     return result;
   }
 
+  private async _toggleDisabledStatus(req: Request, res: Response, actionName: 'enable'|'disable'){
+    const action = {
+      enable: {
+        dbMethod: this._dbManager.enableDocument.bind(this._dbManager),
+        logEvent: this._logEnableDocumentEvents.bind(this),
+        shouldInterrupt: false
+      },
+      disable: {
+        dbMethod: this._dbManager.disableDocument.bind(this._dbManager),
+        logEvent: this._logDisableDocumentEvents.bind(this),
```

Comment:
The log methods look like they could be combined, as do the enable/disableDocument methods, and if they were combined then this method could be simpler?